dot/state/inmemory_storage.go

```diff
@@ -76,9 +76,12 @@ func (s *StorageState) StoreTrie(ts *rtstorage.TrieState, header *types.Header)
 
 	logger.Tracef("cached trie in storage state: %s", root)
 
-	if err := ts.Trie().WriteDirty(s.db); err != nil {
-		logger.Warnf("failed to write trie with root %s to database: %s", root, err)
-		return err
+	// TODO: all trie related db operations should be done in pkg/trie
```

Comment:
same here and for other instances

Reply:
same answer than [here](https://github.com/ChainSafe/gossamer/pull/3768#discussion_r1541503348) it's more a comment to explain the usage of type assertion